### PR TITLE
[SPARK-43772][BUILD][CONNECT] Move version configuration in `connect` module to parent

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to move version configuration in `connect` module to parent.

### Why are the changes needed?
For better management and post maintenance, eg: upgrading some library version of a module while forgetting another one.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Pass GA.
- Manually test
  sh build/sbt "connect/testOnly org.apache.spark.sql.connect.artifact.ArtifactManagerSuite"
  sh build/sbt "connect/testOnly org.apache.spark.sql.connect.planner.SparkConnectServiceSuite"
  sh build/sbt "connect/testOnly org.apache.spark.sql.connect.service.InterceptorRegistrySuite"

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -276,6 +276,13 @@
     <spark.test.docker.keepContainer>false</spark.test.docker.keepContainer>
     <spark.test.docker.removePulledImage>true</spark.test.docker.removePulledImage>
 
+    <!-- Version used in Connect -->
+    <connect.guava.version>31.0.1-jre</connect.guava.version>
```
Comment (panbingkun) [inline pom.xml:280]: Why call `connect.guava.version`? Because the name `guava.version` already exists in this file.

**Comment by panbingkun** [comment]
image

**Comment by panbingkun** [comment]
@zhenlineo @vicennial @LuciferYang this is ready for review.

**Review by zhenlineo — APPROVED** [review]
LGTM

**Comment by LuciferYang** [comment]
@panbingkun please help confirm if the content of the `shaded/assembly` jar?

**Comment by panbingkun** [comment]
> @panbingkun please help confirm if the content of the `shaded/assembly` jar?

The check process is as follows:
1.command: ./build/sbt connect/assembly
2.Before patch:
image

3.After patch:
image

**Comment by panbingkun** [comment]
1.command: ./build/sbt connect-client-jvm/assembly
2.Before patch:
image

3.After patch:
image

**Comment by panbingkun** [comment]
About maven:
1.command: ./build/mvn -pl :spark-connect-client-jvm_2.12 -pl :spark-connect_2.12 -Phadoop-3 -Phive-2.3 -Pyarn -Phive-thriftserver -Phive -DskipTests clean package
2.Before:
image
3.After:
image
4.Diff
image

**Comment by panbingkun** [comment]
1.Before
image
2.After
image
3.Diff
image

**Review by LuciferYang — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged to master. Thanks @panbingkun @zhenlineo

**Pull request closed without merging**